### PR TITLE
[xcode15] Add support for Xcode 15 and corresponding OS versions.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: tests/common/DotNet.cs [hunk 1]
```diff
@@ -176,6 +176,10 @@ public static ExecutionResult Execute (string verb, string project, Dictionary<s
 				// Work around https://github.com/dotnet/msbuild/issues/8845
 				args.Add ("/v:diag");
 				args.Add ("/consoleloggerparameters:Verbosity=Quiet");
+				args.Add ("/consoleloggerparameters:Verbosity=Quiet");
```
Comment (rolfbjarne) [inline tests/common/DotNet.cs:179]: Do we need this argument twice?

**Review by rolfbjarne — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

File: tests/common/DotNet.cs (hunk 1 above)
Comment (mandel-macaque) [reply tests/common/DotNet.cs:179]: I bet is a too fast yy.

File: tests/monotouch-test/CoreText/StringAttributes.cs [hunk 2]
```diff
@@ -28,6 +28,9 @@ public class StringAttributesTests {
 		[Test]
 		public void SimpleValuesSet ()
 		{
+			if (TestRuntime.CheckXcodeVersion (15, 0))
+				Assert.Ignore ("Test timeouts on Xcode 15 beta 4.");
```
[the remark this answers is not in the document]
Comment (mandel-macaque) [reply tests/monotouch-test/CoreText/StringAttributes.cs:32]: I'll add an issue and will link it to the master xcode15 issue

**Comment by dalexsoto** [comment]
Ref: https://github.com/xamarin/xamarin-macios/issues/18556

**Review by dalexsoto — APPROVED** [review]

**Review by haritha-mohan — APPROVED** [review]

File: src/Foundation/Compat.cs [hunk 3]
```diff
@@ -60,6 +61,7 @@ public NSUserActivity ()
 #if __IOS__ && !__MACCATALYST__
 	public partial class NSUrlConnection {
 		// Extension from iOS5, NewsstandKit
+		[EditorBrowsable (EditorBrowsableState.Never)]
```
[the remark this answers is not in the document]
Comment (mandel-macaque) [reply src/Foundation/Compat.cs:64]: There are some of them that are Obsoleted but not removed, that is, they are a warning that in the future they'll be removed. Others, like this ones, are obsoleted and **removed** by apple and so we want to make sure that users don't even know about them.

**Pull request merged**